pkg/controller/backup_finalizer_controller.go

```diff
@@ -201,6 +222,35 @@ func (r *backupFinalizerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 			return ctrl.Result{}, errors.Wrap(err, "error uploading backup final contents")
 		}
 	}
+
+	//Defer Deleting CSI Snapshots
+	// Empty slices here so that they can be passed in to the persistBackup call later, regardless of whether or not CSI's enabled.
+	// This way, we only make the Lister call if the feature flag's on.
+	var volumeSnapshots []snapshotv1api.VolumeSnapshot
+	var volumeSnapshotContents []snapshotv1api.VolumeSnapshotContent
+	if features.IsEnabled(velerov1api.CSIFeatureFlag) {
```

Comment:
I don't see anything else using volumeSnapshots or volumeSnapshotContents slices if bool is false.